### DISCUSSION TITLE
Performance issue

### DISCUSSION THREAD

**Discussion by michabs** [sharkdp/bat#2600] [General]
Comparing bat command to less command on huge file.
Show huge difference when navigating using keystroke of 
"g"- top of file 
"G"- end of file

command used:
$ seq 1 1000000 > tmp.txt
$ less -N tmp.txt
$ bat tmp.txt

**Comment by keith-hall** [comment]
This is to some degree expected - see https://github.com/sharkdp/bat/issues/2218#issuecomment-1238662724

Even though in this case it's plain text, `bat` still passes each line to the syntax highlighting library as for non-plain text files, a prior line could start a block comment etc so there's no way for `bat` to skip lines, whereas a pager can just seek to the end of the file

**Reply by michabs** [reply]
Why not check the first line to determine if "syntax highlighting" required.
May be we need to add a flag that will allow the activation of the bat command in this mode?

**Reply by keith-hall** [reply]
One of `bat`s main features is syntax highlighting. Adding a new code path and complicating the code base (including integration tests, benchmarks etc) for this tiny use case likely isn't worth it. If you want decorations but not syntax highlighting, there are other ways to achieve that without `bat`...
